gpflow/kernels/base.py

```diff
@@ -167,8 +167,8 @@ class Combination(Kernel):
 
     _reduction = None
 
-    def __init__(self, kernels: List[Kernel], name: Optional[str] = None):
-        super().__init__(name=name)
+    def __init__(self, kernels: List[Kernel], **kwargs):
```

Comment:
Can you explain the purpose of the `**kwargs`?

Reply:
https://github.com/GPflow/GPflow/pull/1167#issuecomment-561240349

Reply:
On a wider scale, I do not understand the purpose of this PR

Reply:
@condnsdmatters, btw, I cannot load the link for some reason

Reply:
The purpose of this whole PR is to give abstract parent `MultioutputKernel` a property `output_dims`, that derived classes set at instantiation.  

The reason for kwargs is that we have Multiple Inheritance, with a "diamond structure". That is, some Kernels (eg `SeparateIndependent`) need to inherit from both `Combination` and  `MultioutputKernel`. 

Naively, you would need to explicitly instantiate both parents, with the right arguments. Before - there were no arguments that were uniquely for `MultioutputKernel`; now there are. INstead of explicitly instantiating the parents, the pythonic way to do this is to use `super()` to deal with the MRO and the allow the`__init__` of each class to strip off the arguments it needs. Naturally, all intermediated parent `__init__` methods in this diamond need a kwargs. The children/leafs do not need kwargs. Hope this helps.

Reply:
On that topic, an _excellent_ talk by Raymond Hettinger: [Super considered super!](https://www.youtube.com/watch?v=EiOglTERPEo) (based on this [older blog post](https://rhettinger.wordpress.com/2011/05/26/super-considered-super/))